### DISCUSSION TITLE
(Re-)Generating bakta outputs from the JSON only

### DISCUSSION THREAD

**Discussion by cpauvert** [oschwengers/bakta#300] [Ideas]
Hello!
First of all, a huge thank you for developing and maintaining bakta, I can only imagine the amount of efforts, these are appreciated!

I use bakta regularly and like initially to have all the files (EMBL, faa, ffn, gbff, gff tsv etc.) in case some of my colleagues would like to browse or pipe them to other tools. However, with larger quantities of isolates, larger quantities of files, and I figured, not all of them needs to be backed up as they could all be re generated from the JSON.

The idea would be to have a small scripts that could be bundled with bakta (with the repo only) to import the JSON file and allow to write any of the standards bakta outputs files if they need to be regenerated.

Initially, I thought it might already exists, so I checked around the main bakta repository (and the io files) and the [bakta-ui](https://github.com/ag-computational-bio/bakta-web-ui), but could not find exactly what I was  looking for.

Do you already have something like this in-house, or is it already online and I was blinded?

Keep up the good work!
Best,

**Comment by oschwengers** [comment]
Hi @cpauvert , thanks for reaching out. Excellent question, I also thought about this but never got into action since no one ever asked for it ;-)

I'll have a look into it and potentially put it on my list for an upcoming `v1.10.0` release. Thanks!

**Reply by oschwengers** [reply]
@cpauvert , it took a while, but with #339, Bakta has now an option to reconstruct its output files from the (gzipped) JSON result file. An upcoming `v1.10.0` will be released, soon. Just in case, you're willing to give it an early test upfront, feedback and comments are highly welcome ;-)

**Reply by cpauvert** [reply]
Thanks @oschwengers for taking this seriously and implementing this! Yes I'm happy to try it out.

Unfortunately, I was not able to run it for the moment as it failed with:

```python
Traceback (most recent call last):
  File "/home/cpauvert/miniconda3/envs/bakta-dev/bin/bakta_io", line 33, in <module>
    sys.exit(load_entry_point('bakta', 'console_scripts', 'bakta_io')())
  File "/home/cpauvert/miniconda3/envs/bakta-dev/bin/bakta_io", line 25, in importlib_load_entry_point
    return next(matches).load()
  File "/home/cpauvert/miniconda3/envs/bakta-dev/lib/python3.10/importlib/metadata/__init__.py", line 173, in load
    return functools.reduce(getattr, attrs, module)
AttributeError: module 'bakta.io' has no attribute 'main'

```

When running: `bakta_io --output test.faa test.json` (following indications at https://github.com/oschwengers/bakta/pull/339)

But I realised that I may not have installed this dev version of bakta the right way.

1. clone the repository (fac21b7)
2. `cd bakta`
3. Set the conda environment with the dependencies `mamba create -n bakta-dev -c conda-forge -c bioconda bakta`
4. Activate the environment: `conda (or mamba) activate bakta-dev`
5. Install via local pip to "update" bakta to local dev version: `python -m pip install -e .`


I first went to get answers at https://github.com/oschwengers/bakta/blob/main/CONTRIBUTION.md (which is already nice that it exists, thanks!). But once I figured out a correct way to install a dev bakta, I could update this page.

BEst,

PS: allowing gzipped JSON is a good idea!

**Reply by oschwengers** [reply]
Thanks for the hint and reporting. I could reproduce this. And, sry, my fault. I accidentally named the new module `io.py` just like the existing package `io`. Giving it a unique name solved the issue for me. Fixed by https://github.com/oschwengers/bakta/commit/c47b8f6c6ca47066d2e2e6f6cd0e4ed675389acd

**Reply by oschwengers** [reply]
FYI: You're command list above to create a dev conda env is perfectly fine. I use this as well for local debugging purposes. I added this to the `CONTRIBUTION.md` for others, as well.
